nestauk/asf_installer_survey
# Code section 6

---

# Description

Analysis to generate cross tables from survey data answering each research question from Section 6 of the ASF Installer Survey analysis plan.
- add `data_setup.py`: Pre-processing of dataset and function definitions
- add `free_text_recode.py`: Re-assignment of free text response answers
- add `section_6_analysis.py`: Generating cross table .csv and figure .png outputs
- add new subfolders to `asf_installer_survey/outputs`:
```
└── 📁outputs
    └── 📁section6
        └── 📁csv  # save directory for cross tables
            └── 📁recoded  # save directory for cross tables for research questions with recoded free text responses
        └── 📁figures  # save directory for figures
```

# Instructions for Reviewer

In order to test the code in this PR you need to:
- Clone repo
- See `requirements_section_6.txt` to create environment
- Open 'section_6_analysis.py' using jupytext to open as a Jupyter Notebook and run all cells
-- This should create .csv and .png files for 25 research questions in the relevant `outputs/section6` locations

In your review, please could you:
- Check all cells work for you
- General check for any mistakes/errors in logic (and if you have time, any suggestions for efficiency improvements)
- Check output are generated as expected
-- For further context, outputs are used to create [Flourish figures](https://docs.google.com/presentation/d/10jbXQdzv00t-BaAhltYcDSNlGYt3wpjRHboWGtcVkdg/edit#slide=id.g2cf9e53a955_0_1)


# Checklist:
